pyrefinebio/token.py

```diff
@@ -14,56 +15,81 @@ class Token:
+        >>> token = pyrefinebio.Token(email_address="foo@bar.com")
 
-            email_address (str): the email that the terms and conditions should be sent to.
-        """
+    Create a Token and activate it
 
-        response = post_by_endpoint("token").json()
+        ex:
+        >>> import pyrefinebio
+        >>> token = pyrefinebio.Token(email_address="foo@bar.com")
+        >>> token.agree_to_terms_and_conditions()
 
-        token_id = response["id"]
+    Save a Token to the config file
 
-        if email_address:
-            terms = response["terms_and_conditions"]
-            # email terms
+        ex:
+        >>> import pyrefinebio
+        >>> token = pyrefinebio.Token(email_address="foo@bar.com")
+        >>> token.save_token()
 
-        return token_id
+    Load the Token that is currently saved to the config file
 
-    @classmethod
-    def agree_to_terms_and_conditions(cls, api_token):
+        ex:
+        >>> import pyrefinebio
+        >>> token = pyrefinebio.Token.load_token()
+    """
+
+    def __init__(
+        self,
+        email_address=None,
+        id=None,
+    ):
+        if id:
+            try:
+                response = get_by_endpoint("token/" + str(id))
+            except NotFound:
+                raise BadRequest(
+                    "Token with id '" + str(id) + "' does not exist in RefineBio. " 
+                    "Please create a new token."
+                )
+        else:
+            response = post_by_endpoint(
+                "token",
+                payload={
+                    "email_address": email_address
+                }
+            ).json()
+
+        self.email_address = email_address
+
+        self.token_id = response["id"]
+        self.is_activated = response["is_activated"]
+        self.terms_and_conditions = response["terms_and_conditions"]
+
+
+    def agree_to_terms_and_conditions(self):
         """Activates a token.
 
         Activating a token indicates agreement with refine.bio's
         [Terms of Use](https://www.refine.bio/terms) and
         [Privacy Policy](https://www.refine.bio/privacy).
-
-        parameters:
-
-            api_token (str): the uuid string identifying the token
-                             you want to activate.
         """
-        return put_by_endpoint("token/" + api_token, payload={"is_activated": True})
+        return put_by_endpoint("token/" + self.token_id, payload={"is_activated": True})
 
-    @classmethod
-    def save_token(cls, api_token):
+
+    def save_token(self):
         """Saves a token to the config file.
 
         The default config file is ~/.refinebio.yaml, but
         you can use the environment variable `CONFIG_FILE` to change this path
-
-        parameters:
-
-            api_token (str): the uuid string identifying the token
-                             you want to save.
         """
         config = Config()
-        config.save("token", api_token)
+        config.save("token", self.token_id)
```

Comment:
I'm thinking we might want to add a check here to make sure the token is active before we save it. We don't save `is_activated` for tokens, so I think the assumption is basically that they are activated if you're saving/loading them, so we should enforce that. Besides, why would you really want to save an inactive token?